pkg/util/hostname/kubelet/hostname.go

```diff
@@ -36,10 +37,27 @@ func HostnameProvider(ctx context.Context, options map[string]interface{}) (stri
 		return "", fmt.Errorf("couldn't fetch the host nodename from the kubelet: %s", err)
 	}
 
-	clusterName := clustername.GetClusterName(ctx, nodeName)
+	clusterName := getRFC1123CompliantClusterName(ctx, nodeName)
 	if clusterName == "" {
 		log.Debugf("Now using plain kubernetes nodename as an alias: no cluster name was set and none could be autodiscovered")
 		return nodeName, nil
 	}
-	return (nodeName + "-" + clusterName), nil
+	return nodeName + "-" + clusterName, nil
+}
+
+// getRFC1123CompliantClusterName returns a k8s cluster name if it exists, either directly specified or autodiscovered
+// Some kubernetes cluster-names (EKS,AKS) are not RFC1123 compliant, mostly due to an `_`.
+// This function replaces the invalid `_` with a valid `-`.
+func getRFC1123CompliantClusterName(ctx context.Context, hostname string) string {
+	clusterName := clustername.GetClusterName(ctx, hostname)
+	return makeClusterNameRFC1123Compliant(clusterName)
+}
+
+func makeClusterNameRFC1123Compliant(clusterName string) string {
```

Comment:
does this function can be call several time?
If yes are we ok the send several time the log warning?

Reply:
Yes it will always be called during hostAlias collection. I believe it should warn more than once for clarity? But maybe we can reduce this? Any big opinions?

Reply:
since we replace the value, I don't think it should be a warning. maybe a debug.
because if a user is ok with the renaming, he don't want to have the warning.

maybe it is better to add the information in the `agent status` or in the `agent diagnose` command

Reply:
I can move this to debug because it is unlikely to actually affect a customer directly (due to being an alias we use in the backend). 
But the customer actually doesn't know that we rename this, as we just do this. 


But we need some information for visibility that we do this somewhere, I am not sure which is the correct place either. 
with above change this will already be reflected in the hostAlias section under `agent status`


```
  Hostnames
  =========
    ec2-hostname: ip-172-31-20-143.eu-central-1.compute.internal
    host_aliases: [ip-172-31-20-143.eu-central-1.compute.internal-nam-cluster-test]
    hostname: i-00f1b9e5833c7edc9
    instance-id: i-00f1b9e5833c7edc9
```

Maybe I can add a field to make it clear that we renamed it

```
  Hostnames
  ========= 
    ec2-hostname: ip-172-31-20-143.eu-central-1.compute.internal
    host_aliases: [ip-172-31-20-143.eu-central-1.compute.internal-nam-cluster-test]
    Note: we renamed the hostAlias containing a "_" to "-"
    hostname: i-00f1b9e5833c7edc9
    instance-id: i-00f1b9e5833c7edc9
```

but this doesn't seem to be the best output as well.
So maybe a diagnose? 
Any preference @clamoriniere @mfpierre ?